source/going_further.rst

```diff
@@ -0,0 +1,472 @@
+file and definitions in another:
+
+.. code-block:: python
+    
+    # File polygons.py
+    class Circle:
+        def __init__(self, radius):
+            self.no_sides = 0
+            self.radius = radius
+
+    class Triangle:
+        def __init__(self, side):
+            self.no_sides = 3
+            self.side = side
+
+    class Square:
+        def __init__(self):
+            self.no_sides = 4
+            self.side = side
+
+.. code-block:: python
+
+    # File shapefunctions.py
+    PI = 3.141592 # or use math.PI
+    def calc_perimeter(obj):
+        if obj.no_sides > 0:
+            perimeter = obj.no_sides * obj.side
+        else:  
+            perimeter = 2*PI*obj.radius
+        return perimeter
+
+Importing functionalities
+-------------------------
+
+To use definitions and perimeter function you must now import those functionalities. Code below asks user to specify which shape to
+use and for the side length/radius, and calculates perimeter:
+
+.. code-block:: python
+
+    # Main file or command line
+    import polygons
+    import shapefunctions
+
+    print("Choose a shape:")
+    print("1. Circle")
+    print("2. Triangle")
+    print("3. Square")
+    choice = int(input())
+    print("Choose the side length")
+    side = int(input())
+    if choice == 1:
+        shape = polygons.Circle(side)
+    elif choice == 2:
+        shape = polygons.Triangle(side)
+    elif choice == 3:
+        shape = polygons.Square(side)
+    per = shapefunctions.perimeter(shape)
+    print("The perimeter of your shape is %3.3f" % per)
+
+The ``import polygons`` and ``import shapefunctions`` lines tell the program that we are going to use functionality defined in files ``polygons.py`` and ``shapefunctions.py``.
+Note that these files must be in a place Python can find them, e.g. in the same directory you are running
+python from or on a path defined by ``PYTHONPATH`` variable. Also, note that in the imports we are not using file type extensions, i.e. there is no ``.py`` in
+``import`` statements.
+
+There is also a possibility of using aliases for libraries:
+
+.. code-block:: python
+    
+    # Main file or command line
+    import polygons as plg
+    triangle = plg.Triangle(10)
+
+Here, whenever you need objects from ``polygons`` library, you can use ``plg`` alias instead of writing its full name.
+
+Importing from
+--------------
+
+Sometimes you might want to explicitly state what you are importing. In that case use ``import from`` statement:
+
+.. code-block:: python
+
+    # Main file or command line
+    from polygons import Circle, Triangle
+    triangle = Triangle(10)
+    circle   = Circle(20)
+
+In this case, to create a shape you don't need to precede it's name with the name of the library. This approach, although more explicit,
+and `Explicit is better than implicit <zen-of-python>`_),
```

Comment:
I believe this sentence refers to `from polygons import *` situation ;) 
That being said, I also prefer importing a single thing and not the entire file :+1:

Reply:
I think it's a more general remark. It's here because you can be explicit in various ways. The one presented here, however, has some hidden drawbacks.